intermediate_source/torch_export_tutorial.py

```diff
@@ -0,0 +1,462 @@
+torch._logging.set_logs(dynamic=logging.WARNING, dynamo=logging.WARNING)
+
+######################################################################
+# We can view an ``ExportedProgram``'s constraints using the ``range_constraints`` and
+# ``equality_constraints`` attributes. The logging above reveals what the symbols ``s0, s1, ...``
+# represent.
+
+print(exported_constraints_example3.range_constraints)
+print(exported_constraints_example3.equality_constraints)
+
+######################################################################
+# We can also constrain on individual values in the source code itself using
+# ``constrain_as_value`` and ``constrain_as_size``. ``constrain_as_value`` specifies
+# that a given integer value is expected to fall within the provided minimum/maximum bounds (inclusive).
+# If a bound is not provided, then it is assumed to be unbounded. ``constrain_as_size``
+# is similar to ``constrain_as_value``, except that it should be used on integer values that
+# will be used to specify tensor shapes -- in particular, the value must not be 0 or 1 because
+
+from torch.export import constrain_as_size, constrain_as_value
+
+def constraints_example4(x, y):
+    b = y.item()
+    constrain_as_value(b, 3, 5)
+    if b >= 3:
+       return x.cos()
+    return x.sin()
+
+exported_constraints_example4 = export(constraints_example4, (torch.randn(3, 3), torch.tensor([4])))
+print(exported_constraints_example4(torch.randn(3, 3), torch.tensor([5])))
+try:
+    exported_constraints_example4(torch.randn(3, 3), torch.randn([2]))
+except Exception:
+    tb.print_exc()
+
+def constraints_example5(x, y):
+    b = y.item()
+    constrain_as_size(b)
+    z = torch.ones(b, 4)
+    return x.sum() + z.sum()
+
+exported_constraints_example5 = export(constraints_example5, (torch.randn(2, 2), torch.tensor([4])))
+print(exported_constraints_example5(torch.randn(2, 2), torch.tensor([5])))
+try:
+    exported_constraints_example5(torch.randn(2, 2), torch.randn([1]))
+except Exception:
+    tb.print_exc()
+
+######################################################################
+# Custom Ops
+# ----------
+# ``torch.export`` can export PyTorch programs with custom operators.
+#
+# NOTE: the API for registering custom ops is still under active development
+# and may change without notice.
+#
+# Currently, the steps to register a custom op for use by ``torch.export`` are:
+# - Define the custom op using ``torch.library`` as with any other custom op
+
+from torch.library import Library, impl
+
+m = Library("my_custom_library", "DEF")
+
+m.define("custom_op(Tensor input) -> Tensor")
+
+@impl(m, "custom_op", "CompositeExplicitAutograd")
+def custom_op(x):
+    print("custom_op called!")
+    return torch.relu(x)
+
+# - Define a ``Meta`` implementation of the custom op that returns an empty
+# tensor with the same shape as the expected output
+
+@impl(m, "custom_op", "Meta")
```

Comment:
Side note of naming: since the company is called "Meta", I feel this name might be misinterpreted.
Perhaps an all lowercase "meta" would be a little bit better?

Reply:
I believe using the exact key "Meta" is required in order to get this working - this is a dispatch-related thing.